### PR TITLE
Consolidate list-like checking to is_list_like

### PR DESCRIPTION
**Pull request by max-sixty**
Closes https://github.com/pydata/pandas/issues/11740

Let me know what tests would be helpful for these types of 'cleaning' PRs, and whether a "what's new" is needed.

### CODE REVIEW

**Comment by MaximilianR** [comment]
@kawochen green. 
In the end, I'm not sure it's much simpler, although it is a bit more encapsulated & explicit
Let me know your thoughts

File: pandas/core/common.py [hunk 1]
```diff
@@ -2016,14 +2016,15 @@ def _asarray_tuplesafe(values, dtype=None):
 
 
 def _index_labels_to_array(labels):
-    if isinstance(labels, (compat.string_types, tuple)):
+    from .index import Index # ABCIndex doesn't work here
```
Comment (jreback) [inline pandas/core/common.py]: I *think* you might be able to eliminate this routine (``_index_labels_to_array``) entirely, its only in a couple of places

*and* I think this is overlapping a couple of other routines which do this (you'll have to look a bit in the code)

**Comment by jreback** [comment]
@MaximilianR update?

**Comment by MaximilianR** [comment]
Yup, on it. Aiming for this w/e.

**Comment by jreback** [comment]
closing, but pls reopen if working on it

**Comment by MaximilianR** [comment]
OK, apologies for letting this drop.

**Comment by jreback** [comment]
np again. your contributions are great! just trying to move things forward.

**Pull request closed without merging**